CMakePresets.json

```diff
@@ -0,0 +1,28 @@
+{
+    "version": 3,
+    "configurePresets": [
+        {
+            "name": "macOS base",
+            "hidden": true,
+            "binaryDir": "${sourceDir}/build/",
```

Comment:
I would suggest removing this? AFAIK this has a reasonable default in the global vscode cmake settings, and I believe this will override a users own settings, which could cause other problems in their build environment.

Reply:
I personally think the `binaryDir` is one of the best aspects of CmakePresets - no need to execute `mkdir -p build && cd build` anymore.
Again, presets are fully optional to use, they are opinionated and provide sensible defaults which should cover most setups.

But I don't understand the remark regarding vscode cmake settings? This file is actually independent of vscode.